File: oops/aa_SubObj.cpp
#include <iostream>
using namespace std;

class Student{
  public:
  int id;
  char name;

  void insert(int i, char n){
    id = i;
    name = n; 
  }

void display(){
  cout << "ID: " << id << ", Name: " << name << endl;
 }
};

int main() {
  Student s1;
  Student s2;                 // Static
  Student *s3=new Student;    // Dynamic obj

  s1.insert(1, 'A');
  s2.insert(2, 'B');
  s3->insert(3, 'C');
//  (*s3)->insert(4, 'D');
  
  s1.display();
  s2.display();
  s3->display();
  return 0;
}

File: oops/aa_SubObj.java
class Student {
  public int id;
  public char name;

  public void insert(int i, char n) {
    id = i;
    name = n;
  }

  public void display() {
    System.out.println("ID: " + id + ", Name: " + name);
  }
}

public class aa_SubObj {
  public static void main(String[] args) {
    Student s1 = new Student();       // Static object
    Student s2 = new Student();       // Static object
    Student s3 = new Student();       // Dynamic object (like C++'s `new`)

    s1.insert(1, 'A');
    s2.insert(2, 'B');
    s3.insert(3, 'C');
    // There's no equivalent for `(*s3)->insert(4, 'D');` in Java

    s1.display();
    s2.display();
    s3.display();
  }
}
